packages/vite/src/node/plugins/resolve.ts

```diff
@@ -461,7 +461,12 @@ function tryFsResolve(
     return res
   }
 
-  for (const ext of options.extensions) {
+  // if fspath from node_modules, should use DEFAULT_EXTENSIONS (#11467).eg
+  const extensions = fsPath.includes('node_modules')
+    ? DEFAULT_EXTENSIONS
+    : options.extensions
```

Comment:
I think this is a sensible default change, but if the user specifies extra extensions not included in `DEFAULT_EXTENSIONS`, we need to add them too, e.g. `DEFAULT_EXTENSIONS.concat(extraExtensionsNotInDefault)`.

It might be a good idea to precompute this outside of the plugin object too.

Reply:
it's a good idea, a more comprehensive fix, thx!